torch/utils/throughput_benchmark.py

```diff
@@ -0,0 +1,87 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import torch._C
+
+class ThroughputBenchmark(object):
+    '''
+    This class is a wrapper around a c++ component throughput_benchmark::ThroughputBenchmark
+    responsible for executing a PyTorch module (nn.Module or ScriptModule)
+    under an inference server like load. It can emulate multiple calling threads
+    to a single module provided. In the future we plan to enhance this component
+    to support inter and intra-op parallelism as well as multiple models
+    running in a single process.
+
+    Please note that even though nn.Module is supported, it might incur an overhead
+    from the need to hold GIL every time we execute Python code or pass around
+    inputs as Python objects. As soon as you have a ScriptModule version of your
+    model for inference deployment it is better to switch to using it in this
+    benchmark.
+
+    Example::
+
+        >>> from torch.utils import ThroughputBenchmark
+        >>> bench = ThroughputBenchmark(my_module)
+        >>> # Pre-populate benchmark's data set with the inputs
+        >>> for input in inputs:
+            # Both args and kwargs work, same as any PyTorch Module / ScriptModule
+            bench.add_input(input[0], x2=input[1])
+        >>> Inputs supplied above are randomly used during the execution
+        >>> stats = bench.benchmark(
+                num_calling_threads=4,
+                num_warmup_iters = 100,
+                num_iters = 1000,
+            )
+        >>> print("Avg latency (ms): {}".format(stats.latency_avg_ms))
+        >>> print("Number of iterations: {}".format(stats.num_iters))
```

Comment:
Do we really believe that this heavy OOP API is the right way to go? Why can't the benchmark utility be something like this? It looks much simpler and fits it all in a single line.
```py
stats = torch.utils.bench.throughput(module, args=(x, y), kwargs=dict(a=1, b=2), num_calling_threads=4, ...)
```

Reply:
Additionally, aren't we concerned that running the same model over and over on the same input data might cause artifacts in the benchmark due to the input data remaining in cache?

Reply:
1. We need our current API to populate the dataset first 
2. We are not running over the same input . Depending on the number of inputs provided we will have more or less cache hits. Ideally you provide a dataset which doesn't fit your l3 before running benchmark function